pkg/apis/camel/v1/build_types.go

```diff
@@ -41,6 +41,8 @@ type BuildSpec struct {
 	// and its phase set to BuildPhaseFailed.
 	// +kubebuilder:validation:Format=duration
 	Timeout metav1.Duration `json:"timeout,omitempty"`
+	// the maximum amount of parallel running builds started by this operator instance
+	MaxRunningBuilds int32 `json:"maxRunningBuilds,omitempty"`
```

Comment:
I think the Build should not be aware of this information. It seems something to the domain of the controller instead, not of this resource specifically.